wpilibj/src/sim/java/edu/wpi/first/wpilibj/Preferences.java

```diff
@@ -675,7 +675,7 @@ public void valueChanged(ITable source, String key, Object value, boolean isNew)
           synchronized (lock) {
             if (!ImproperPreferenceKeyException.isAcceptable(key) || value.toString().indexOf
                 ('"') != -1) {
-              if (values.contains(key) || keys.contains(key)) {
+              if (values.containsValue(key) || keys.contains(key)) {
```

Comment:
This is incorrect.  We want to see if the key is a key in either hashtable here.